normandy/recipes/api/views.py

```diff
@@ -88,14 +89,48 @@ def update(self, request, *args, **kwargs):
         Intercept PUT requests and have them create instead of update
         if the object does not exist.
         """
-        if request.method == 'PUT':
+        if request.method in ['PUT', 'PATCH']:
             try:
-                self.get_object()
+                recipe = self.get_object()
             except Http404:
-                return self.create(request, *args, **kwargs)
+                if request.method == 'PUT':
+                    return self.create(request, *args, **kwargs)
+            else:
+                if recipe.is_approved:
+                    recipe.disable(ignore_revision_id=True)
```

Comment:
this does unapprove and disable it:
https://github.com/mozilla/normandy/pull/131/files#diff-c5447b2cc20ceb64379f180e6f3a5d00R130